spikeinterface/toolkit/qualitymetrics/pca_metrics.py

```diff
@@ -333,64 +344,82 @@ def nearest_neighbors_isolation(waveform_extractor: si.WaveformExtractor, this_u
     """
 
     rng = np.random.default_rng(seed=seed)
-
-    all_units_ids = waveform_extractor.sorting.get_unit_ids()
-    other_units_ids = np.setdiff1d(all_units_ids, this_unit_id)
-
-    # get waveforms for target cluster
-    waveforms_target_unit = waveform_extractor.get_waveforms(unit_id=this_unit_id)
-    n_spikes_target_unit = waveforms_target_unit.shape[0]
-
-    # find units whose closest channels overlap with closest channels of target cluster
-    closest_chans_all = get_template_channel_sparsity(waveform_extractor, method='radius',
-                                                      outputs='index', peak_sign='both',
-                                                      radius_um=radius_um)
-    closest_chans_target_unit = closest_chans_all[this_unit_id]
-    other_units_ids = [unit_id for unit_id in other_units_ids if \
-        np.any(np.in1d(closest_chans_all[unit_id], closest_chans_target_unit))]
-
-    # if no unit is within neighborhood of target unit, then just say isolation is 1 (best possible)
-    if not other_units_ids:
-        nn_isolation = 1
-
-    # if there are units to compare, then compute isolation with each
+    
+    sorting = waveform_extractor.sorting
+    all_units_ids = sorting.get_unit_ids()
+    n_spikes_all_units = compute_num_spikes(waveform_extractor)
```

Comment:
@samuelgarcia I ended up using the `compute_num_spikes` from the `quality_metrics` module. Will make a separate PR for the core function